zilliqa/src/consensus.rs

```diff
@@ -306,23 +306,59 @@ impl Consensus {
             self.view = view;
             self.block_store.request_block_by_view(view)?;
         }
-        self.view = to + 1;
+
+        if to > self.view {
+            info!(
+                "*** Downloading missing blocks...view is going from {} to {}",
+                self.view,
+                to + 1
+            );
+            self.view = to + 1;
+        }
 
         Ok(())
     }
 
-    pub fn timeout(&mut self) -> Result<(PeerId, NewView)> {
+    pub fn timeout(&mut self, peers_joined: bool) -> Option<(PeerId, ExternalMessage)> {
+        if self.view == 1 {
+            if self.config.allow_single_node_network || peers_joined {
+                let genesis = self
+                    .get_block_by_view(0)
+                    .unwrap()
+                    .ok_or_else(|| anyhow!("missing block"))
+                    .unwrap();
+                // If we're in the genesis committee, vote again.
+                if genesis
+                    .committee
+                    .iter()
+                    .any(|v| v.peer_id == self.peer_id())
+                {
+                    warn!("timeout in view 1, we should vote for genesis block rather than incrementing view");
```

Comment:
I don't think this should be a warning, since it will happen whenever you start a new network (unless I'm misunderstanding?)